### DISCUSSION TITLE
Features & ideas for the next release

### DISCUSSION THREAD

**Discussion by dkhd** [dkhd/winterly-tab#17] [Ideas]
Hi @sano2019 & @je-poy 

I mentioned in this discussion since you both are currently the only contributors in this project :)))

As you already know (and hopefully install it as well), Winterly Tab is already released for Chrome, Edge, and Firefox.

I'm thinking about adding new features for the next release on the extension store, and I need your opinion on these.

## What about adding support to personalize the Winterly Tab? 

Here's what I proposed:

- Something like **light & dark mode**. This dark mode is implemented by adding a dark layer above the image when activated, preferably using `nord0` or similar color, while the light mode is removing (or just changing the opacity to `0`) the darkener layer. I choose to use Nord theme since we are in-line with what Nord theme provides. Nord theme is basically a set of color palette with arctic & north-bluish theme (maybe we need to put some information about it as well in the README.md. Winterly Tab & Nord theme would be a very icy & good match. More about Nord theme can be read in [nordtheme.com](http://nordtheme.com/).

- Customize the clock. It is either the time format, 12H/24H, or clock location. Current clock is locked in bottom-left side and cannot be dynamically changed.

- Input field to put the users' name. It will be used to greet the users whenever they open the new tab.

- Adding analytics to track the behavior. Well, I know this will be debatable, but I think we need to track the behavior for this project future improvement (no need to track the personal information). Since this project is open source, we can also use something like open source analytic tools like [Plausible](https://plausible.io/) or [Mamoto](https://matomo.org/).

All those personalization above will utilize the `localStorage` of the browser. So the data will not leave the users' browser (and we do not need to build any backend service at this stage).

What do you think, guys? I'm willing to hear your opinion on this matters.

Have a nice snowy day! And may your coffee always hot ☕

**Comment by je-poy** [comment]
Hello @dkhd,

These features are great, this will make the extension more personalized.
With this, does this means will be having a simple form panel for customization?

Here are my suggestions:
- For the light and dark mode, we can use the device default theme as the default winter tab theme. A toggle button would be also good for switching between themes.

- For the time format, when the clock component is clicked, a pop up options will appear where the user can choose from the given formats.

- For the Users' name, still I am not sure if there are data stored/cached on the browser where we can automatically get the user info. From there, instead of inputting the users' name manually, we can retrieve it from that data storage.

- For the analytics, this is a nice-to-have feature.  I also checked the sites, they have trial periods. Plausible doesn't have free tier, only 30 days free unlimited trial. On the other hand, Momoto also have free trial for their cloud service however they have Mamoto On-Premise which is free. The cons for this on-premise set up is that we will be maintaining backend servers for us to serve the analytics service and data storage, which is I guess, costly... I'll try to search other options.


Hoping your coffee is still hot hoho, btw there is no snow here  ☀️ 
Have a great day ahead.

**Comment by dkhd** [comment]
Thanks for the response, @je-poy 

```
For the light and dark mode, we can use the device default theme as the default winter tab theme. A toggle button would be also good for switching between themes.
```
This would be a great feature! By default, Winterly Tab will respect the OS theme, and users have choice to change it either by a toggle button or from the settings.

---
```
For the time format, when the clock component is clicked, a pop up options will appear where the user can choose from the given formats.
```
So, the configuration is not from the settings section?

---
```
For the analytics, this is a nice-to-have feature. I also checked the sites, they have trial periods. Plausible doesn't have free tier, only 30 days free unlimited trial. On the other hand, Momoto also have free trial for their cloud service however they have Mamoto On-Premise which is free. The cons for this on-premise set up is that we will be maintaining backend servers for us to serve the analytics service and data storage, which is I guess, costly... I'll try to search other options.
```
Yeah, this is just an optional addition for now, no need to rush it :) And maybe it's also good for the users to opt-out from this tracking (like a toggle button).

---

By the way for the settings, I'm thinking about utilizing the same dialog you just created, so the users don't need to leave the page.


Yeah, there are no snow as well here in Indonesia 😄 But I would love to experience it someday if I have the chance.

**Reply by je-poy** [reply]
> So, the configuration is not from the settings section?

I think we can disregard the initial suggestion I made for the time format, let us just use the settings panel for all the setups.

<hr/>

> By the way for the settings, I'm thinking about utilizing the same dialog you just created, so the users don't need to leave the page.

Yeap, let us separate it from the About dialog box, however.

**Comment by dkhd** [comment]
Hi @je-poy 

I merged the PR from @boidushya to add search bar. Hi @boidushya 👋 make yourself at home.

@je-poy this is interesting, we can add multiple widgets into Winterly Tab. At least that currently we have clock and search bar, and these addition will add new idea to put things to the settings page.

**Reply by je-poy** [reply]
@boidushya, Nice! Welcome :)

Sure @dkhd , that would be fun 👍

**Reply by boidushya** [reply]
Hey! Thanks for the feedback!
Might be a stretch but what if we let users configure which widgets to keep and which to purge?

**Comment by bigsbunny** [comment]
Hello everyone! I am new here!

I have a few suggestions that can be a part of Winterly (apart from the Quotes functionality I mentioned earlier):

1. We can have a section where we can store the user's most frequently visited links to websites, similar to any web browser's stock homepage. But we can keep it more subtle.
2. Instead of keeping the clock at the bottom-left of the page, we can place it at the center with a larger font-size so that it is a bit more explicit.
3. Instead of text at the bottom-right of the page (fullscreen, about, change background), we can replace them with **icons** pointing to the feature.

These are some things I have in mind! Happy to be a part of this :)

**Comment by dkhd** [comment]
Hi @bigsbunny 👋

Welcome to the family 😄 we are more than happy to have you here

About your suggestions, they are amazing! Maybe we can start implementing the settings so users can personalize Winterly the way they want?

I would be very helpful for the user if they can configure it and your suggestion would be very helpful.

Again, welcome!

**Reply by bigsbunny** [reply]
Yeah that's a better idea actually. Instead of providing the features by default, we can provide the user with options to customize Winterly to their own liking.

**Comment by je-poy** [comment]
Good day, i created some of points here as Feature Request so any can see what are we needing.
